lib/amp/plugin.js

```diff
@@ -94,16 +94,19 @@ const createCustomHead = originalHead => function customHead () {
    */
   ensureKey(head, 'link', [])
 
-  if (origin && !head.link.find(l => l.rel === 'canonical' || l.hid === 'canonical')) {
-    const path = this.$isAMP && this.$ampMode !== 'only'
-      ? this.$route.fullPath.replace(/^\/amp(\/.*)?/, '$1')
-      : this.$route.fullPath
-
-    head.link.push({
-      rel: 'canonical',
-      hid: 'canonical',
-      href: origin + path
-    })
+  // Add canonical meta only if page is served as AMP
+  if (this.$ampMode !== false && this.$isAMP === true) {
```

Comment:
Checking `this.$isAMP` is enough to detect AMP pages

Reply:
Indeed it is!